src/test/java/net/snowflake/client/jdbc/ProxyLatestIT.java

```diff
@@ -0,0 +1,411 @@
+                            "java",
+                            "-jar",
+                            getWiremockStandAlonePath(),
+                            "--root-dir",
+                            System.getProperty("user.dir")
+                                + File.separator
+                                + WIREMOCK_HOME_DIR
+                                + File.separator,
+                            "--enable-browser-proxying", // work as forward proxy
+                            "--proxy-pass-through",
+                            "false", // pass through only matched requests
+                            "--port",
+                            String.valueOf(httpProxyPort),
+                            "--https-port",
+                            String.valueOf(httpsProxyPort),
+                            "--https-keystore",
+                            getResourceURL("wiremock" + File.separator + "ca-cert.jks"),
+                            "--ca-keystore",
+                            getResourceURL("wiremock" + File.separator + "ca-cert.jks"))
+                        .start();
+                waitForWiremock();
+                return true;
+              } catch (Exception e) {
+                logger.info(
+                    "Failed to start wiremock, retrying: " + Arrays.toString(e.getStackTrace()));
+                return false;
+              }
+            });
+  }
+
+  private static void downloadWiremock() {
+    try (CloseableHttpClient client = HttpClients.createDefault();
+        CloseableHttpResponse response = client.execute(new HttpGet(WIREMOCK_STANDALONE_URL))) {
+      HttpEntity entity = response.getEntity();
+      File wiremockStandaloneFile = new File(getWiremockStandAlonePath());
+      wiremockStandaloneFile.getParentFile().mkdirs();
+      wiremockStandaloneFile.createNewFile();
+      try (FileOutputStream outputStream = new FileOutputStream(wiremockStandaloneFile, false)) {
+        entity.writeTo(outputStream);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static String getWiremockStandAlonePath() {
+    return System.getProperty("user.dir")
+        + File.separator
+        + WIREMOCK_HOME_DIR
+        + File.separator
+        + WIREMOCK_FILE_NAME;
+  }
+
+  private void waitForWiremock() {
+    await()
+        .pollDelay(Duration.ofSeconds(1))
+        .atMost(Duration.ofSeconds(5))
+        .until(this::isWiremockResponding);
+  }
+
+  private boolean isWiremockResponding() {
+    try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+      HttpGet request =
+          new HttpGet(String.format("http://%s:%d/__admin/mappings", WIREMOCK_HOST, httpProxyPort));
+      CloseableHttpResponse response = httpClient.execute(request);
+      return response.getStatusLine().getStatusCode() == 200;
+    } catch (Exception e) {
+      logger.warning("Waiting for wiremock to respond: " + Arrays.toString(e.getStackTrace()));
```

Comment:
`e.printStackTrace` is not good enough?

Reply:
this will print log in one line which I preferred to not extend logs too much (but we can change if you think `e.printStackTrace` would be better)

Reply:
@sfc-gh-dprzybysz chime in, please :)

Reply:
Instead we can just pass the exception to the logger and the logger would be responsible for formatting the message

Reply:
done